bin/build-darwin

```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+GOOS=darwin GOARCH=amd64 go build -v -buildmode=c-shared -o proxy/planetscale-darwin-amd64.so
+GOOS=darwin GOARCH=arm64 go build -v -buildmode=c-shared -o proxy/planetscale-darwin-arm64.so
+
+lipo -create -output proxy/planetscale-darwin.so proxy/planetscale-darwin-amd64.so  proxy/planetscale-darwin-arm64.so
```

Comment:
Xcode provides `lipo` which afaik should be available on Actions too to build this single universal file.